File: Tool/FMDB/FMDatabaseQueueManager.m
//
//  FMDatabaseQueueManager.m
//  X8.2O
//
//  Created by 巴糖 on 2016/12/20.
//  Copyright © 2016年 巴糖. All rights reserved.
//

#import "FMDatabaseQueueManager.h"
#import "NSString+Sandbox.h"
#import "FMDB.h"

static FMDatabaseQueueManager *_manager;

@interface FMDatabaseQueueManager()<NSCopying>

@property (nonatomic, strong) FMDatabaseQueue *databaseQueue;

@end

@implementation FMDatabaseQueueManager

+ (instancetype)shareManager{
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        _manager = [[FMDatabaseQueueManager alloc] init];
    });
    return _manager;
}

+ (instancetype)allocWithZone:(struct _NSZone *)zone{
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        _manager = [[FMDatabaseQueueManager allocWithZone:zone] init];
    });
    return _manager;
}

- (id)copyWithZone:(NSZone *)zone{
    return _manager;
}

- (void)opendatabaseQueue:(NSString *)databaseQueueName{
    NSString *path = databaseQueueName.wx_documentsDir;
    self.databaseQueue = [FMDatabaseQueue databaseQueueWithPath:path];
}

- (void)closedatabaseQueue{
    [self.databaseQueue close];
}

- (void)createTableDatabaseQueue:(NSArray *)sqls{
    NSString *sql = [sqls componentsJoinedByString:@""];
    [self.databaseQueue inDatabase:^(FMDatabase *db) {
        [db executeStatements:sql];
    }];
}
@end

File: Objective-C/Tool/FMDB/FMDatabaseQueueManager.swift
//
//  FMDatabaseQueueManager.swift
//
//
//  Created by 巴糖 on 2016/12/19.
//  Copyright © 2016年 巴糖. All rights reserved.
//

import UIKit

class FMDatabaseQueueManager: NSObject {
    
    class func shareManager() -> FMDatabaseQueueManager { return manager }
    var dbQueue: FMDatabaseQueue?
}

/** FMDatabaseQueue **/
extension FMDatabaseQueueManager{
    func openDBQueue(_ DBName: String) {
        let path = DBName.documentsDir()
        dbQueue = FMDatabaseQueue(path: path)
    }
    
    func closeDBQueue() {
        dbQueue?.close()
    }
    
    func createTableDBQueue(sqls: [String]){
        guard  let dbQueue = FMDatabaseQueueManager.shareManager().dbQueue else {
            return
        }
        dbQueue.inDatabase { (db) in
            for sql in sqls {
                db?.executeUpdate(sql, withArgumentsIn: nil)
            }
        }
    }
}

extension FMDatabaseQueueManager{
    fileprivate static let manager: FMDatabaseQueueManager = FMDatabaseQueueManager()
}
